src/jit/lsraarm64.cpp

```diff
@@ -365,16 +365,46 @@ void LinearScan::TreeNodeInfoInit(GenTree* tree)
             break;
 
         case GT_CMPXCHG:
-            info->srcCount = 3;
+        {
+            GenTreeCmpXchg* cmpXchgNode = tree->AsCmpXchg();
+            info->srcCount              = cmpXchgNode->gtOpComparand->isContained() ? 2 : 3;
             assert(info->dstCount == 1);
 
-            // TODO-ARM64-NYI
-            NYI("CMPXCHG");
-            break;
+            info->internalIntCount = 1;
+
+            // For ARMv8 exclusives the lifetime of the addr and data must be extended because
+            // it may be used used multiple during retries
+            cmpXchgNode->gtOpLocation->gtLsraInfo.isDelayFree = true;
+            cmpXchgNode->gtOpValue->gtLsraInfo.isDelayFree    = true;
+            if (!cmpXchgNode->gtOpComparand->isContained())
+            {
+                cmpXchgNode->gtOpComparand->gtLsraInfo.isDelayFree = true;
+            }
+            info->hasDelayFreeSrc = true;
+
+            // Internals may not collide with target
+            info->isInternalRegDelayFree = true;
```

Comment:
@CarolEidt might want to take a look and approve this section:

```
the lifetime of the addr and data must be extended because
 it may be used used multiple during retries
```